src/mpid/ch4/shm/posix/posix_coll.h

```diff
@@ -256,9 +261,29 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_POSIX_mpi_allreduce(const void *sendbuf, void
     goto fn_exit;
 
   fallback:
+    if (MPIR_CVAR_ENABLE_YAKSA_REDUCTION) {
+        is_supported = MPIR_Typerep_reduce_is_supported(op, datatype);
+    }
+    if (!is_supported) {
+        MPIR_Coll_host_buffer_alloc(sendbuf, recvbuf, count, datatype, &host_sendbuf,
```

Comment:
Same comment apply here. But on a second thought, you don't need this `is_supported` variable at all.  You could do
```
if (MPIR_Typerep_reduce_is_supported(...) {
    host_sendbuf = NULL;
    host_recvbuf = NULL;
} else {
    MPIR_Coll_host_buffer_alloc(...);
}
```
, and leave the rest of the code intact, right?